src/weights.jl

```diff
@@ -650,21 +650,29 @@ function quantile(v::AbstractVector{V}, w::AbstractWeights{W}, p::AbstractVector
     vw = sort!(collect(zip(view(v, nz), view(w, nz))))
     N = length(vw)
 
+    # missing is always sorted last
+    if ismissing(vw[end][1])
+        throw(ArgumentError("quantiles are undefined in presence of missing values"))
+    end
+
     # prepare percentiles
     ppermute = sortperm(p)
     p = p[ppermute]
 
     # prepare out vector
-    out = Vector{typeof(zero(V)/1)}(undef, length(p))
+    v1 = vw[1][1]
+    out = Vector{typeof(v1 + zero(eltype(p))*zero(W)*zero(v1))}(undef, length(p))
```

Comment:
Ideally I wanted to use `map(p)` and return the result directly to avoid having to compute the type manually, but I couldn't find a way for the return type to be inferred. The problem is that the loop does not just depend on the quantile, we also have to update the sum of weights, current and previous values, which get boxed. Not sure there's a clean solution.